### ISSUE TITLE
Heads-up on upstreaming

### ISSUE THREAD

**Issue by CoMPaTech** [laetificat/anna-ha#10]
Hi @laetificat, @bouwew (and indirectly @plamola),

I asked some questions on HASS discord and some of the other developers indicate that we should just have a go at upstreaming. They also recommend to start early and not including all bells and whistles. I've prepped https://github.com/CoMPaTech/home-assistant/tree/anna-ha-0.9 which is basically the code as we already have it running, altered to fit the coding styles required.

I'd like to put in a first PR somewhere in the coming days (aiming at wednesday evening to submit it) just to see how far we can get with upstreaming. This would be without the operation mode and async @laetificat indicated in my first PR but this is in line with what other devs recommend (small steps).

**Comment by laetificat** [comment]
Sounds good to me, but does this mean the main repo for this component will be the home assistant one or this one?

**Comment by bouwew** [comment]
Yes, this component, when accepted, is added to the HA-repo, with assigned maintainer(s).
But, I think there is nothing against keeping this repo alive for development and testing purposes? With a small change to the name, to differentiate it from the HA-component, there is no conflict I would think.

**Comment by CoMPaTech** [comment]
Sorry, missed the updates on this issue :(

Indeed would seem like a good approach to test and break things ourselves. Mopping up my commits for 0.96 to this repo as we speak.

**Comment by CoMPaTech** [comment]
If we keep it here as anna-ha (although for broader testing eventually) and upstream it as plugwise (as already now in the component)? We can change it back to anna, or even anna-ha as a custom component name so we can better align with updates of haanna, right?

**Comment by CoMPaTech** [comment]
PRs submitted, expecting a decent amount of feedback from HA :)
“Hold on, it’s going to be a bumpy right with lots of learning curves and turbulent user enthousiasm”

Waiting on https://github.com/laetificat/haanna/issues/7

PRs:
https://github.com/home-assistant/home-assistant/pull/25533
https://github.com/home-assistant/home-assistant.io/pull/9992

**Comment by bouwew** [comment]
@CoMPaTech, as long as the names are different, it should work, the only issue could be the double communication to the Smile. But I think HA manages that internally somehow...

**Comment by CoMPaTech** [comment]
I suggest we rename it accordingly here to plugwise-dev or something likewise :) But I do thing custom_components tend to override/overload the core ones

**Comment by CoMPaTech** [comment]
As of yesterday #25533 is merged into dev 👍 
If no objections or improvements, I'll rework the code here as plugwise-dev for the `custom_component` so we can have a go at looking at Adams etc. that also provides us with a working `entity` and the one we are developing against

**Comment by bouwew** [comment]
Great!!
I just loaded the code from #25533 into a custom component.

**Comment by CoMPaTech** [comment]
HA #25533 and HA.io #9922 are merged ... we didn't make it to 0.97 but we will be part of 0.98  👍 

Currently rebasing this `custom_component` from those PRs - and yes they play together nicely
![plugwise-dev-front](https://user-images.githubusercontent.com/5751088/62369414-dbf70480-b52f-11e9-9e2a-5b20d4a93920.png)
![plugwise-dev-states](https://user-images.githubusercontent.com/5751088/62369415-dbf70480-b52f-11e9-967f-e330a6791b81.png)
